### PR TITLE
2021 04 28 azure first draft for revision

### PR DESCRIPTION
**Pull request by smaffulli**

### CODE REVIEW

File: _posts/2021-04-28-aks-cost.md [hunk 1]
```diff
@@ -0,0 +1,388 @@
...
+tags:
+  - cost monitoring
+  - Kubecost
+  - AKS
+  - Azure
+---
+
+# Monitoring and Governance for AKS Costs with Kubecost
+
+#### How to calculate cluster efficiency and health of Azure Kubernetes Service
```
Comment (dwbrown2) [inline _posts/2021-04-28-aks-cost.md:21]: This subtitle feels misleading to me. Can we remove or update?

File: _posts/2021-04-28-aks-cost.md [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+
+The Microsoft [Azure Kubernetes Service](https://azure.microsoft.com/en-us/services/kubernetes-service/) (AKS) is a [popular](https://security.stackrox.com/rs/219-UEH-533/images/State_of_Container_and_Kubernetes_Report.pdf) fully managed Kubernetes service that offers embedded continuous integration and continuous delivery as well as enterprise-grade security and governance— powerful tools for teams adopting Kubernetes. As with any complex infrastructure, AKS requires proper governance and financial transparency for successful organizational adoption. Kubecost, an open source tool that provides teams with visibility into Kubernetes spend and supports environments hosted in Azure, is a widely recommended solution for engineers and finance teams facing this problem. _Note: This [documentation page](https://docs.microsoft.com/en-us/azure/cloud-adoption-framework/scenarios/aks/eslz-security-governance-and-compliance#cost-governance) for AKS provides helpful context for using Kubecost to implement a cost governance strategy. _
```
Comment (dwbrown2) [inline _posts/2021-04-28-aks-cost.md:25]: This italics formatting does not appear to be working when I view here:

https://github.com/kubecost/blog/blob/78cdc416a977392c263a5fa4b41b7c991d9f99c9/_posts/2021-04-28-aks-cost.md

File: _posts/2021-04-28-aks-cost.md [hunk 3]
```diff
@@ -0,0 +1,388 @@
...
+
+
+1. **Dedicated AKS cluster**: Each team or project gets its own dedicated AKS cluster to easily track its billing. This approach does not benefit from resource sharing. \
+
+2. **Dedicated AKS node**: Each team or project gets its [pods assigned to a specific node](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/) in a shared AKS cluster using labels and a configuration constraint known as nodeSelector. Each team can then isolate its spending in the Azure bill according to the cost of its assigned nodes. Resource sharing is not efficient in this model. \
+
+3. **Dedicated AKS Namespace**: Each team or project gets its own dedicated namespace along with more flexibility and administrative freedom. This model, as we have mentioned, can be challenging for cost allocation. However, this is where Kubecost comes in.  
+
+
+![With Kubecost, you can share AKS clusters and still clearly allocate shared costs.](../assets/images/2021-04-28-aks-cost/image12.png "With Kubecost, you can share AKS clusters and still clearly allocate shared costs.")
```
Comment (dwbrown2) [inline _posts/2021-04-28-aks-cost.md:50]: The and most other images are really large. It makes the article hard to read IMO.

File: _posts/2021-04-28-aks-cost.md [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+This means that you must allocate the following by namespace:
+
+1. The actual usage of CPU, GPU, memory, network, and persistent storage.
+2. The excessive requests for CPU and memory leaving resources unused.  
+3. The usage of shared resources such as databases or blob storage.
+4. The proportionate use of cloud instance reservations and support services.
+
+![An accurate cost allocation by project, team, or cost center must consider all factors.](../assets/images/2021-04-28-aks-cost/image5.png "An accurate cost allocation by project, team, or cost center must consider all factors.")
+
+FIXME CAPTION An accurate cost allocation by project, team, or cost center must consider all factors.
```
Comment (dwbrown2) [inline _posts/2021-04-28-aks-cost.md:68]: What's the plan with these? I don't believe our GH blog supports but Medium does.

Comment (smaffulli) [reply _posts/2021-04-28-aks-cost.md:66]: we can either try something like this https://stackoverflow.com/questions/19331362/using-an-image-caption-in-markdown-jekyll (that requires some testing) or we remove them. I think they add something but if it's too hard to fool around with Jekyll let's drop them.

File: _posts/2021-04-28-aks-cost.md [hunk 5]
```diff
@@ -0,0 +1,386 @@
+---
+title: "Free AKS Cost Governance With Kubecost"
+description: "Kubecost allows Azure AKS administrators to allocate and
+report on costs, calculate cluster efficiency and health, and alert on
+a budget excess."
+date: 2021-04-28T11:34:00-04:00
+canonical_url: "http://blog.kubecost.com/blog/aks-cost/"
+author: Michael Dresser
```
Comment (smaffulli) [inline _posts/2021-04-28-aks-cost.md:8]: who shall be the author?

File: _posts/2021-04-28-aks-cost.md (hunk 4 above)
Comment (dwbrown2) [reply _posts/2021-04-28-aks-cost.md:66]: I vote drop them for now, merge, and then revisit if we have time. Your call though.

File: _posts/2021-04-28-aks-cost.md (hunk 5 above)
Comment (dwbrown2) [reply _posts/2021-04-28-aks-cost.md:8]: Ajay Tripathy or Sean Holcomb?`

**Pull request merged**